docs/implementation/pruning.md

```diff
@@ -0,0 +1,82 @@
+## Pruning deployments
+
+Pruning is an operation that deletes data from a deployment that is only
```

Comment:
I might start with the higher level context for, maybe something like...
By default, subgraphs store a full version history for entities, allowing consumers to query the subgraph as of any historical block.  Pruning is an operation that deletes entity versions from a deployment older than a certain block, so it is no longer possible to query that deployment as of prior blocks. In GraphQL...

Reply:
Nice, incorporated.